spec/rubocop/cop/rails/timecop_spec.rb

```diff
@@ -0,0 +1,177 @@
+              .to(eq('freeze_time'))
+          end
+        end
+      end
+
+      context 'with arguments' do
+        it 'adds an offense' do
+          expect_offense(<<-RUBY.strip_indent)
+            Timecop.freeze(123)
+            ^^^^^^^^^^^^^^^^^^^ Use `travel` or `travel_to` instead of `Timecop.freeze`
+          RUBY
+        end
+
+        it 'does not autocorrect' do
+          source = 'Timecop.freeze(123)'
+
+          expect(autocorrect_source(source)).to(eq(source))
+        end
+      end
+    end
+
+    context 'with a block' do
+      context 'without arguments' do
+        it 'adds an offense' do
+          expect_offense(<<-RUBY.strip_indent)
+            Timecop.freeze { }
+            ^^^^^^^^^^^^^^ Use `freeze_time` instead of `Timecop.freeze`
+          RUBY
+        end
+
+        it 'autocorrects to `freeze_time`' do
+          expect(autocorrect_source('Timecop.freeze { }'))
+            .to(eq('freeze_time { }'))
+        end
+      end
+
+      context 'with arguments' do
+        it 'adds an offense' do
+          expect_offense(<<-RUBY.strip_indent)
+            Timecop.freeze(123) { }
+            ^^^^^^^^^^^^^^^^^^^ Use `travel` or `travel_to` instead of `Timecop.freeze`
+          RUBY
+        end
+
+        it 'does not autocorrect' do
+          source = 'Timecop.freeze(123) { }'
+
+          expect(autocorrect_source(source)).to(eq(source))
+        end
+      end
+    end
+  end
+
+  describe 'Timecop.return' do
+    context 'without a block' do
+      it 'adds an offense' do
+        expect_offense(<<-RUBY.strip_indent)
+          Timecop.return
+          ^^^^^^^^^^^^^^ Use `travel_back` instead of `Timecop.return`
+        RUBY
+      end
+
+      context 'in Rails < 6.0', :rails5 do
+        it 'autocorrects to `travel_back`' do
+          expect(autocorrect_source('Timecop.return')).to(eq('travel_back'))
+        end
+
+        context 'inside a block' do
+          it 'autocorrects to `travel_back`' do
+            expect(autocorrect_source('foo { Timecop.return }'))
+              .to(eq('foo { travel_back }'))
+          end
+        end
+      end
+
+      # context 'in Rails > 6.0', :rails6 do
+      #   it 'autocorrects to `unfreeze`' do
+      #     expect(autocorrect_source('Timecop.return')).to(eq('unfreeze'))
+      #   end
+
+      #   context 'inside a block' do
+      #     it 'autocorrects to `unfreeze`' do
+      #       expect(autocorrect_source('foo { Timecop.return }'))
+      #         .to(eq('foo { unfreeze }'))
+      #     end
+      #   end
+      # end
+    end
+
+    context 'with a block' do
+      it 'adds an offense' do
+        expect_offense(<<-RUBY.strip_indent)
+          Timecop.return { }
+          ^^^^^^^^^^^^^^ Use `travel_back` instead of `Timecop.return`
+        RUBY
+      end
+
+      it 'does not autocorrect' do
+        expect(autocorrect_source('Timecop.return { }'))
+          .to(eq('Timecop.return { }'))
+      end
+
+      context 'inside a block' do
```

Comment:
Is this really necessary to cover this case? I.e. it's detected in any context.

Reply:
This edge case can cause false positives for `was_passed_block?` and forces this change:

```diff
 def was_passed_block?(node)
   node.send_type? && node.parent &&
-    node.parent.block_type?
+    node.parent.block_type? && node.parent.send_node == node
 end
```

This is because the in both the cases of

```ruby
foo { Timecop.return }
# (block
#   (send nil :foo)
#   (args)
#   (send (const nil :Timecop) :return)   ←
# )
```

and
```ruby
Timecop.return { assert true }
# (block
#   (send (const nil :Timecop) :return)   ←
#   (args)
#   (send nil :assert (true))
# )
```

the parent node of

```ruby
Timecop.return
#   (send (const nil :Timecop) :return)
```

is a `block` node, so we must make sure to distinguish between the two. Hence testing: 
- passed no block
- passed no block, in a block
- passed a block
- passed a block, in a block